src/features/device-logs/lib/config.ts

```diff
@@ -1,21 +1,47 @@
 import type { DeviceLogsBackend, LogContext } from './struct.js';
 
 import _ from 'lodash';
-import { RedisBackend } from './backends/redis.js';
 import {
 	LOGS_DEFAULT_RETENTION_LIMIT,
-	LOKI_QUERY_HOST,
 	LOKI_READ_PCT,
-	LOKI_INGESTER_HOST,
 	LOKI_WRITE_PCT,
+	LOGS_LOKI_ENABLED,
+	LOGS_REDIS_ENABLED,
+	LOGS_PRIMARY_BACKEND,
+	LOGS_REDIS_READ_PCT,
+	LOGS_REDIS_WRITE_PCT,
 } from '../../../lib/config.js';
 
-export const LOKI_ENABLED =
-	LOKI_QUERY_HOST && LOKI_INGESTER_HOST && LOKI_WRITE_PCT > 0;
-export const shouldPublishToLoki = () =>
-	LOKI_ENABLED && LOKI_WRITE_PCT > Math.random() * 100;
-export const shouldReadFromLoki = () =>
-	LOKI_ENABLED && LOKI_READ_PCT > Math.random() * 100;
+const LOGS_SECONDARY_BACKEND =
+	LOGS_PRIMARY_BACKEND === 'loki' ? 'redis' : 'loki';
+
+export const LOGS_SECONDARY_BACKEND_ENABLED =
+	LOGS_SECONDARY_BACKEND === 'loki'
+		? LOGS_LOKI_ENABLED
+		: LOGS_SECONDARY_BACKEND === 'redis'
+			? LOGS_REDIS_ENABLED
+			: false;
```

Comment:
NIT: Isn't this just v ?
```suggestion
export const LOGS_SECONDARY_BACKEND_ENABLED =
	(LOGS_SECONDARY_BACKEND === 'loki' && LOGS_LOKI_ENABLED) ||
	(LOGS_SECONDARY_BACKEND === 'redis' && LOGS_REDIS_ENABLED);
```

Reply:
sure

Reply:
Your suggestions fails linting with
```
  19:59  error  Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator  @typescript-eslint/prefer-nullish-coalescing

Reply:
Fixed it